### PR TITLE
[CECO-941][docs] Add introspection docs

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Add public docs about introspection

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/introspection.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Introspection (beta)
+
+This feature was released in operator v1.4.0 and is currently in beta.
+
+## Overview
+
+Introspection allows the operator to detect a node's environment and automatically make configuration changes based on it. Each environment is referred to as a `provider`. Examples include GKE Container-Optimized OS (GKE COS), Azure Kubernetes Service (AKS), and Red Hat OpenShift. Depending on the node's provider, the Datadog Agent on that node may require certain configurations be set to run properly. Introspection creates a Datadog Agent deployment for each provider which includes provider-specific configurations needed to run the Agent and a provider-specific node affinity.
```
Comment (hestonhoffman) [inline docs/introspection.md:7]:
```suggestion
Introspection allows the operator to detect a node's environment and automatically make configuration changes based on it. Each environment is referred to as a `provider`. Examples include GKE Container-Optimized OS (GKE COS), Azure Kubernetes Service (AKS), and Red Hat OpenShift. Depending on the node's provider, the Datadog Agent on that node may require certain configurations be set to run properly. Introspection creates a Datadog Agent deployment for each provider, which includes provider-specific configurations needed to run the Agent and a provider-specific node affinity.
```

File: docs/introspection.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Any node that does not have an associated provider will have a `default` provider apply to them. The `default` provider does not contain any special configuration.
```
Comment (hestonhoffman) [inline docs/introspection.md:9]:
```suggestion
Any node that does not have an associated provider will have a `default` provider applied to them. The `default` provider does not contain any special configuration.
```

**Review by hestonhoffman — APPROVED** [review]
Couple small edits

File: docs/introspection.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Introspection is disabled by default. To enable introspection using the [datadog-operator helm chart](https://github.com/DataDog/helm-charts/tree/main/charts/datadog-operator), set `introspection.enabled=true` in your `values.yaml` file or as a flag in the command line arguments `--set introspection.enabled=true`.
+
+**Note:** Enabling introspection in an existing operator installation will create new DaemonSets or ExtendedDaemonSets without deleting the existing DaemonSet or ExtendedDaemonSet. The existing DaemonSet or ExtendedDaemonSet needs to be removed manually. This behavior will change in a future version of the operator to make the migration process smoother.
+
+## Supported Providers
+
+| Provider | Operator Version |
+| -------- | :--------------: |
+| GKE Container-Optimized OS | v1.4.0 |
```
Comment (fanny-jiang) [inline docs/introspection.md:36]:
```suggestion
| GKE v1.24+ Container-Optimized OS | v1.4.0 |
```

Comment (khewonc) [reply docs/introspection.md:36]: Discussed offline. Providing tested versions instead

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**